src/clients/base-server-client.js

```diff
@@ -0,0 +1,160 @@
+'use strict'
+
+const param = require('jquery-param')
+const { isString } = require('../utils/typecheck')
+const simpleFetch = require('../utils/simple-fetch')
+const ServiceTokenClient = require('../service-token/server')
+
+/**
+ * @class Client
+ * @classdesc Basic REST client that can be instantiated for different models.
+ *
+ * @throws Will raise error if baseUrl, doorkeeperBaseUrl, accessKey or secretKey is invalid
+ */
+class Client {
+  constructor (options = {}) {
+    this.name = options.name
+    this.baseUrl = options.url
+
+    this.config = {
+      accessKey: options.accessKey,
+      secretKey: options.secretKey,
+      doorkeeperBaseUrl: options.doorkeeperBaseUrl,
+      authorizationRequired: true,
```

Comment:
ok not sure if it makes sense - should we make`authorizationRequired` part of `options`?